Src/main.c

```diff
@@ -103,13 +103,37 @@ int main(void)
 
   HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
   /* USER CODE END 2 */
+  //Initialize ADC and TIM
+  HAL_ADC_Init(&hadc);
+  HAL_TIM_PWM_Start(&htim16, TIM_CHANNEL_1);
 
   /* Infinite loop */
   /* USER CODE BEGIN WHILE */
+  //set servo PWM output relative to input voltage from potentiometer.
+  //potentiometer voltage from 0-3.3V, PWM modulation from 1-2ms on-time at 50Hz (period = 20ms)
+  
+  int Ch1 = 0; //no need to constantly re-initialize a variable.
   while (1)
   {
+      Ch1 = 0;
+      //ADC conversion:
+      double OnTime = HAL_ADC_GetValue(&hadc); //RAW ADC input
+      OnTime = OnTime / (1023 / 3.3); //Converting raw to input voltage value for clarity (between 0V and 3.3V)
+      OnTime = (OnTime / 3.3) + 1; //Converting to On-time of the PWM by getting the decimal of input Voltage and adding 1ms 
+      while (Ch1 <= OnTime) {
+          TIM16->CCR1 = Ch1;
+          HAL_GPIO_WritePin(GPIOA, GPIO_PIN_6, GPIO_PIN_SET);
+          Ch1 += 1;
+          HAL_Delay(1);
+      }
+      while (Ch1 > OnTime) {//check for both max period and initial value
+          TIM16->CCR1 = Ch1;
+          HAL_GPIO_WritePin(GPIOA, GPIO_PIN_6, GPIO_PIN_RESET);
+          Ch1 += 1;
+          HAL_Delay(1);
+      }
```

Comment:
I'm not sure if you quite understand the purpose of the CCR register in this case. 
http://www.se.rit.edu/~swen-563/slides/STM32%20GPIO%20and%20Timers.pdf 
Page 29 has a nice annotation that I think will help.

Also, just a note with your logic, be wary of employing HAL_Delay or while - if any other pieces of code are dependent on this piece of code, the call is considered blocking and can mess up a lot of stuff inadvertently. Try to stay away from them unless you absolutely need to use them :)

Reply:
That source is very helpful. I was using math for a different mode.